### DISCUSSION TITLE
Calculation of cut_size

### DISCUSSION THREAD

**Discussion by benvekedim** [networkx/networkx#5929] [Q&A] [answered]
I created a fully connected graph on X data.
How can I give weight as distance between two points in cut_size method? 

```
import math
import numpy as np
from sklearn.metrics import pairwise_distances
import networkx as nx
from scipy.spatial import distance

X = np.array([[0,math.sqrt(99)/10],[0.1,0],[5,math.sqrt(99)/10]])
constant = 1

 ```
G is undirected graph.
```
G = nx.Graph()

for node_i in range(len(X)):
  G.add_node(node_i,pos=X[node_i])

A = np.exp(np.divide(-np.power(pairwise_distances(X),2),2*constant)

rows,cols = np.where((A > 0)&(A != 0))

for node_i,node_j in zip(rows,cols):
    G.add_edge(node_i,node_j)


G.remove_edges_from(nx.selfloop_edges(G))

```

I tried to change the weight of the edges.But my cut size result is 1.

```

for i in G.edges(data=True):
  G[i[0]][i[1]]["weight"] = distance.euclidean(X[i[0]],X[i[1]])

S = {0}
T = {2}
nx.cut_size(G,S,T,weight=5)
#output 1 

```

**Comment by brocla** [comment, marked as answer]
Hello @benvekedim. I spent some time looking at your code. I think the issue is the `weight` keyword in the last line. It should read,

```python
nx.cut_size(G,S,T,weight='weight')
```

The keyword in `cut_size` wants to know which edge attribute has the weights.
I don't have `sklearn` installed, so I couldn't check it.
Good Luck

**Reply by benvekedim** [reply]
Thank you for help. Yes it's works :)
